hibernate-core/src/main/java/org/hibernate/event/internal/DefaultMergeEventListener.java

```diff
@@ -455,6 +460,26 @@ protected void copyValues(
 				target,
 				copyCache
 		);
+		EntityMetamodel entityMetamodel = persister.getEntityMetamodel();
+		final boolean enhancedForLazyLoading = PersistentAttributeInterceptable.class.isAssignableFrom( entity.getClass() );
+		final BytecodeEnhancementMetadata enhancementMetadata = entityMetamodel.getBytecodeEnhancementMetadata();
+		final LazyAttributesMetadata lazyAttributesMetadata = enhancementMetadata.getLazyAttributesMetadata();
+
+		if(enhancedForLazyLoading) {
+			final BytecodeLazyAttributeInterceptor interceptor = enhancementMetadata.extractLazyInterceptor(entity);
+			if (interceptor == null) {
+
+				final int span = entityMetamodel.getPropertySpan();
+				final String[] propertyNames = entityMetamodel.getPropertyNames();
+
+				for ( int j = 0; j < span; j++ ) {
+					final String propertyName = propertyNames[j];
+					if(lazyAttributesMetadata.isLazyAttribute( propertyName ) && copiedValues[j] == null) {
+						copiedValues[j] = LazyPropertyInitializer.UNFETCHED_PROPERTY;
+					}
```

Comment:
Keep in mind that this code passes 10,472 of the 10,473 test cases, so no changes are being ignored.  This routine is only called from two places, and both are following cascade.  The code in this locaiton was written for PersistentAttributeInterceptable objects, and was later duplicated in AbstractEntityTuplizer when I worked on the other type of proxy, so I think I can remove it here since the other logic happens first, and in both cases.

Reply:
Yes, actually all the changes in this file have been removed due to being obsolete by later changes in other classes.